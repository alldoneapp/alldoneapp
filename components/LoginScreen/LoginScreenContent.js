import React, { useEffect, useState } from 'react'
import { StyleSheet, View, ImageBackground, Image, Text, ScrollView } from 'react-native'

import Icon from '../Icon'
import styles from '../styles/global'
import LogInButton from '../UIControls/LogInButton'
import URLSystem, { URL_LOGIN } from '../../URLSystem/URLSystem'
import Colors from '../../Themes/Colors'
import { getProjectData, logEvent, loginWithGoogleWebAnonymously } from '../../utils/backends/firestore'
import { useSelector, useDispatch } from 'react-redux'
import { setInitialUrl, setNavigationRoute } from '../../redux/actions'

const LOADING_LOGIN = 'LOADING_LOGIN'
const NORMAL_LOGIN = 'NORMAL_LOGIN'
const GUIDE_LOGIN = 'GUIDE_LOGIN'

export default function LoginScreenContent() {
    const dispatch = useDispatch()
    const initialUrl = useSelector(state => state.initialUrl)
    const [templateTitle, setTemplateTitle] = useState('')
    const [templateImage, setTemplateImage] = useState('')
    const [loginType, setLoginType] = useState(
        window.location.pathname.startsWith('/login') || window.location === '/' ? NORMAL_LOGIN : LOADING_LOGIN
    )

    const getProjectIdFromUrl = url => {
        return url.split('/')[2]
    }

    const selectPictureToShow = async url => {
        await loginWithGoogleWebAnonymously()

        const projectId = getProjectIdFromUrl(url)

        if (projectId) {
            const project = await getProjectData(projectId)

            const isTemplate = project && project.isTemplate
            if (isTemplate) {
                setTemplateTitle('Sign up/in with your personal email')
                setTemplateImage(require('../../web/images/illustrations/guidesLogin.png'))
            }
            setLoginType(isTemplate ? GUIDE_LOGIN : NORMAL_LOGIN)
        } else {
            setLoginType(NORMAL_LOGIN)
        }
    }

    const writeBrowserURL = () => {
        URLSystem.push(URL_LOGIN)
    }

    useEffect(() => {
        let url = ''
        if (initialUrl === '/') {
            const { pathname, search } = window.location
            url = pathname.startsWith('/login') || pathname === '/' ? '/projects/tasks/open' : pathname + search
            dispatch(setInitialUrl(url))
        }
        url ? selectPictureToShow(url) : setLoginType(NORMAL_LOGIN)
    }, [])

    useEffect(() => {
        writeBrowserURL()
        dispatch(setNavigationRoute('LoginScreen'))
        logEvent('login_page')
    }, [])

    return (
        <ImageBackground source={require('../../web/images/illustrations/LoginBg.svg')} style={localStyles.container}>
            <ScrollView showsVerticalScrollIndicator={false} style={localStyles.scrollContainer}>
                <View style={localStyles.scrollSubContainer}>
                    <a href="/" style={{ top: 64, textDecoration: 'none' }}>
                        <View style={{ flex: 1, flexDirection: 'row' }}>
                            <Icon size={24} name={'logo'} color={Colors.Primary100} />
                            <Icon style={localStyles.logoText} size={24} name={'logo-name'} color={Colors.Primary100} />
                        </View>
                    </a>
                    {loginType === NORMAL_LOGIN && (
                        <Image
                            source={{ uri: require(`../../web/images/illustrations/PersonajeEnseÃ±ando.png`) }}
                            style={{ width: 362 * 0.75, height: 470 * 0.75 }}
                            width={362}
                            height={470}
                        />
                    )}
                    {loginType === GUIDE_LOGIN && (
                        <Image
                            source={{ uri: templateImage }}
                            style={{ marginTop: 16, width: 600 * 0.51, height: 692 * 0.51 }}
                            width={600}
                            height={692}
                        />
                    )}

                    {loginType === NORMAL_LOGIN && (
                        <Text style={localStyles.title}>Sign in with your personal email to access Alldone</Text>
                    )}
                    {loginType === GUIDE_LOGIN && <Text style={localStyles.title}>{templateTitle}</Text>}
                    <LogInButton />
                    <Text style={localStyles.subtitle}>
                        Alldone helps you to achieve your goals in life. When you sign in you accept the{' '}
                        <a
                            href="https://alldone.app/privacy"
                            target={'_blank'}
                            style={{ color: Colors.Primary100, textDecoration: 'none' }}
                        >
                            Privacy Policy
                        </a>{' '}
                        and the{' '}
                        <a
                            href="https://alldone.app/terms"
                            target={'_blank'}
                            style={{ color: Colors.Primary100, textDecoration: 'none' }}
                        >
                            Terms of Service
                        </a>{' '}
                        and want to login with your Google account
                    </Text>
                </View>
            </ScrollView>
        </ImageBackground>
    )
}

const localStyles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
    },
    scrollContainer: {
        flex: 1,
        width: '100%',
    },
    scrollSubContainer: {
        flex: 1,
        paddingTop: 64,
        paddingBottom: 32,
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
    },
    title: {
        ...styles.title4,
        textAlign: 'center',
        color: Colors.Text02,
        padding: 16,
        maxWidth: 572,
    },
    subtitle: {
        ...styles.body1,
        textAlign: 'center',
        color: Colors.Text01,
        paddingTop: 32,
        paddingBottom: 16,
        maxWidth: 548,
        paddingHorizontal: 16,
    },
    logoText: {
        marginLeft: 9,
    },
})
