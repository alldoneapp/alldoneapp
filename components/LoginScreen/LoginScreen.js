import React from 'react'
import { useSelector } from 'react-redux'

import LoadingScreen from '../LoadingScreen'
import LoginScreenContent from './LoginScreenContent'

export default function LoginScreen() {
    const addingUserToGuide = useSelector(state => state.addingUserToGuide)
    const loggedIn = useSelector(state => state.loggedIn)
    const registeredNewUser = useSelector(state => state.registeredNewUser)

    return loggedIn ? (
        <LoadingScreen
            text={addingUserToGuide ? 'Preparing your community project...' : 'Loading your info...'}
            secondText={registeredNewUser || addingUserToGuide ? 'Please wait - this can take a while' : ''}
        />
    ) : (
        <LoginScreenContent />
    )
}
