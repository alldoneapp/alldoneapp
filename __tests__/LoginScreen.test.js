import React from 'react'
import LoginScreen from '../components/LoginScreen'
import store from '../redux/store'
import { Provider } from 'react-redux'

import renderer from 'react-test-renderer'
import { setOnline } from '../redux/actions'

jest.mock('../utils/backends/firestore')

jest.mock('../components/MyPlatform', () => {
    return {
        __esModule: true,
        default: {
            isMobile: () => false,
        },
    }
})

jest.mock('../utils/BackendBridge', () => {
    let innerValues = {
        isNewUser: true,
    }
    return {
        __esModule: true,
        default: {
            loginWithGoogle: () => {
                return { additionalUserInfo: { isNewUser: innerValues.isNewUser } }
            },
            uploadNewUser: () => {},
            getProjectsByUser: () => {},
            setInnerValue: x => {
                innerValues = x
            },
            mapUserData: () => {},
        },
    }
})

const initialState = {
    user: {
        uid: '1',
        displayName: 'Dummy Doe',
        photoURL: 'https://somewhereInTheInternet.com/any.png',
    },
    loggedIn: true,
    online: true,
    initialUrl: 'alldone.aleph.engineering/tasks',
}

describe('LoginScreen component', () => {
    store.dispatch(setOnline(true))
    describe('method test', () => {
        xit('after call componentDidMount should render correctly', () => {
            const tree = renderer.create(<LoginScreen />)
            const instance = tree.getInstance()
            instance.componentDidMount()
            expect(tree.toJSON()).toMatchSnapshot()
        })

        it('should navigate to App after projects load', async () => {
            const tree = renderer.create(<LoginScreen navigation={{ navigate: () => {} }} />)
            const instance = tree.getInstance()
            instance.afterProjectsLoad()
            expect(tree.toJSON()).toMatchSnapshot()
        })
    })
})

jest.mock('../redux/store', () => {
    let innerState = {
        user: {
            uid: '1',
            displayName: 'Dummy Doe',
            photoURL: 'https://somewhereInTheInternet.com/any.png',
        },
        loggedIn: true,
        online: true,
    }
    return {
        __esModule: true,
        default: {
            subscribe: x => {},
            unsubscribe: () => {},
            getState: () => {
                return {
                    loggedIn: innerState.loggedIn,
                    loggedUser: innerState.user,
                    online: innerState.online,
                }
            },
            dispatch: x => {
                if (x.type) return
                innerState = x
            },
        },
        foo: jest.fn(() => 43),
    }
})

//This component imports implicitly react-native-gesture-handler, a specific
//Jest setup is required for Jest to test it properly.
//See https://www.gitmemory.com/issue/kmagiera/react-native-gesture-handler/344/489547513 and
//https://stackoverflow.com/questions/57595093/cannot-read-property-direction-of-undefined-tests-only
describe('LoginScreen component', () => {
    describe('LoginScreen snapshot test', () => {
        xit('should render correctly', () => {
            store.dispatch(initialState)
            const tree = renderer
                .create(
                    <Provider store={store}>
                        <LoginScreen />
                    </Provider>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        })

        xit('should render correctly', () => {
            store.dispatch({
                user: {
                    uid: '1',
                    displayName: 'Dummy Doe',
                    photoURL: 'https://somewhereInTheInternet.com/any.png',
                },
                loggedIn: true,
                online: false,
                initialUrl: 'alldone.aleph.engineering/tasks',
            })
            const tree = renderer
                .create(
                    <Provider store={store}>
                        <LoginScreen />
                    </Provider>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()

            store.dispatch(initialState)
        })
    })

    describe('LoginScreen snapshot test', () => {
        xit('should load user projects', () => {
            const tree = renderer
                .create(
                    <Provider store={store}>
                        <LoginScreen />
                    </Provider>
                )
                .toJSON()

            expect(tree).toMatchSnapshot()
        })
    })

    describe('LoginScreen snapshot test', () => {
        it('should render when there is no user logged', () => {
            store.dispatch({
                loggedIn: false,
                online: true,
                user: {
                    uid: '1',
                    displayName: 'Dummy Doe',
                    photoURL: 'https://somewhereInTheInternet.com/any.png',
                },
            })
            const tree = renderer
                .create(
                    <Provider store={store}>
                        <LoginScreen />
                    </Provider>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        })
    })

    describe('method test', () => {
        it('should update the state', () => {
            const tree = renderer.create(<LoginScreen />)

            const instance = tree.getInstance()

            instance.updateState()
        })

        it('should unmount the without errors', () => {
            const tree = renderer.create(<LoginScreen />)
            tree.getInstance().state.unsubscribe = () => {}
            tree.unmount()
        })
    })
})
