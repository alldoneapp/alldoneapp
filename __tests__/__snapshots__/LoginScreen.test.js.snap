// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen component LoginScreen snapshot test should load user projects 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0A44A5",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 168,
        "justifyContent": "space-around",
        "width": 168,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "alldone",
            "fontSize": 69,
            "opacity": 1,
          }
        }
      >
        
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "alldone",
            "fontSize": 35,
            "opacity": 1,
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 4,
          "flexDirection": "row",
          "height": 40,
          "opacity": 1,
          "paddingLeft": 19,
          "paddingRight": 19,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "gray",
              "fontFamily": "alldone",
              "fontSize": 20,
              "opacity": 1,
            }
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Roboto-Medium",
            "fontSize": 14,
            "letterSpacing": 0.8,
            "lineHeight": 14,
            "opacity": 0.54,
            "paddingLeft": 15,
          }
        }
      >
        Enter in Offline Mode
      </Text>
    </View>
  </RNGestureHandlerButton>
</View>
`;

exports[`LoginScreen component LoginScreen snapshot test should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 10000,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#E7ECEF",
          "height": 200,
          "maxHeight": 200,
          "maxWidth": 200,
          "minHeight": 200,
          "minWidth": 200,
          "width": 200,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#0A44A5",
              "fontFamily": "alldone",
              "fontSize": 150,
              "opacity": 1,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "fontFamily": "Roboto-Regular",
        "fontSize": 30,
        "paddingTop": 40,
        "textAlign": "justify",
      }
    }
  >
    Loading your info...
  </Text>
</View>
`;

exports[`LoginScreen component LoginScreen snapshot test should render correctly 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0A44A5",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 168,
        "justifyContent": "space-around",
        "width": 168,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "alldone",
            "fontSize": 69,
            "opacity": 1,
          }
        }
      >
        
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "alldone",
            "fontSize": 35,
            "opacity": 1,
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 4,
          "flexDirection": "row",
          "height": 40,
          "opacity": 1,
          "paddingLeft": 19,
          "paddingRight": 19,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "gray",
              "fontFamily": "alldone",
              "fontSize": 20,
              "opacity": 1,
            }
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Roboto-Medium",
            "fontSize": 14,
            "letterSpacing": 0.8,
            "lineHeight": 14,
            "opacity": 0.54,
            "paddingLeft": 15,
          }
        }
      >
        Enter in Offline Mode
      </Text>
    </View>
  </RNGestureHandlerButton>
</View>
`;

exports[`LoginScreen component LoginScreen snapshot test should render when there is no user logged 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0A44A5",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 168,
        "justifyContent": "space-around",
        "width": 168,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "alldone",
            "fontSize": 69,
            "opacity": 1,
          }
        }
      >
        
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "alldone",
            "fontSize": 35,
            "opacity": 1,
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 4,
          "flexDirection": "row",
          "height": 40,
          "opacity": 1,
          "paddingLeft": 19,
          "paddingRight": 19,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={25}
        bbWidth={25}
        focusable={false}
        height={25}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            undefined,
            Object {
              "opacity": 1,
            },
            Object {
              "flex": 0,
              "height": 25,
              "width": 25,
            },
          ]
        }
        vbHeight={545}
        vbWidth={538}
        width={25}
      >
        <RNSVGGroup
          fill={4278190080}
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        >
          <RNSVGPath
            d="M533.5 278.4c0-18.5-1.5-37.1-4.7-55.3H272.1v104.8h147c-6.1 33.8-25.7 63.7-54.4 82.7v68h87.7c51.5-47.4 81.1-117.4 81.1-200.2z"
            fill={4282549748}
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
          <RNSVGPath
            d="M272.1 544.3c73.4 0 135.3-24.1 180.4-65.7l-87.7-68c-24.4 16.6-55.9 26-92.6 26-71 0-131.2-47.9-152.8-112.3H28.9v70.1c46.2 91.9 140.3 149.9 243.2 149.9z"
            fill={4281641043}
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
          <RNSVGPath
            d="M119.3 324.3c-11.4-33.8-11.4-70.4 0-104.2V150H28.9c-38.6 76.9-38.6 167.5 0 244.4l90.4-70.1z"
            fill={4294687748}
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
          <RNSVGPath
            d="M272.1 107.7c38.8-.6 76.3 14 104.4 40.8l77.7-77.7C405 24.6 339.7-.8 272.1 0 169.2 0 75.1 58 28.9 150l90.4 70.1c21.5-64.5 81.8-112.4 152.8-112.4z"
            fill={4293542709}
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Roboto-Medium",
            "fontSize": 14,
            "letterSpacing": 0.8,
            "lineHeight": 14,
            "opacity": 0.54,
            "paddingLeft": 15,
          }
        }
      >
        Enter with Google
      </Text>
    </View>
  </RNGestureHandlerButton>
</View>
`;

exports[`LoginScreen component method test after call componentDidMount should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 10000,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#E7ECEF",
          "height": 200,
          "maxHeight": 200,
          "maxWidth": 200,
          "minHeight": 200,
          "minWidth": 200,
          "width": 200,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#0A44A5",
              "fontFamily": "alldone",
              "fontSize": 150,
              "opacity": 1,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "fontFamily": "Roboto-Regular",
        "fontSize": 30,
        "paddingTop": 40,
        "textAlign": "justify",
      }
    }
  >
    Loading your info...
  </Text>
</View>
`;

exports[`LoginScreen component method test after call onPressEnterButton should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 10000,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#E7ECEF",
          "height": 200,
          "maxHeight": 200,
          "maxWidth": 200,
          "minHeight": 200,
          "minWidth": 200,
          "width": 200,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#0A44A5",
              "fontFamily": "alldone",
              "fontSize": 150,
              "opacity": 1,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "fontFamily": "Roboto-Regular",
        "fontSize": 30,
        "paddingTop": 40,
        "textAlign": "justify",
      }
    }
  >
    Loading your info...
  </Text>
</View>
`;

exports[`LoginScreen component method test after call onPressEnterButton should render correctly when is not a new user 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 10000,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#E7ECEF",
          "height": 200,
          "maxHeight": 200,
          "maxWidth": 200,
          "minHeight": 200,
          "minWidth": 200,
          "width": 200,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#007FFF",
              "fontFamily": "alldone",
              "fontSize": 150,
              "opacity": 1,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "fontFamily": "Roboto-Regular",
        "fontSize": 30,
        "marginBottom": 30,
        "paddingTop": 40,
        "textAlign": "justify",
      }
    }
  >
    Loading your info...
  </Text>
</View>
`;

exports[`LoginScreen component method test should navigate to App after projects load 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 10000,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#E7ECEF",
          "height": 200,
          "maxHeight": 200,
          "maxWidth": 200,
          "minHeight": 200,
          "minWidth": 200,
          "width": 200,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#007FFF",
              "fontFamily": "alldone",
              "fontSize": 150,
              "opacity": 1,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "fontFamily": "Roboto-Regular",
        "fontSize": 30,
        "marginBottom": 30,
        "paddingTop": 40,
        "textAlign": "justify",
      }
    }
  >
    Loading your info...
  </Text>
</View>
`;
